src/building/warehouse.c

```diff
@@ -115,6 +115,28 @@ int building_warehouse_add_resource(building *b, int resource)
     return 1;
 }
 
+int building_warehouses_add_resource(int resource, int amount, int respect_settings)
+{
+    if (amount <= 0) {
+        return 0;
+    }
+    
+    int remaining_to_add = amount;
+    for (building *b = building_first_of_type(BUILDING_WAREHOUSE); b; b = b->next_of_type) {
+        int continue_while = (remaining_to_add > 0);
```

Comment:
Like granaries, don't forget to check for valid warehouses.

Reply:
Done.
Added:
```
if (b->state != BUILDING_STATE_IN_USE) {
            continue;
        }
        ```